### PR TITLE
Loosen version requirement to allow resque v2.0 and up.

### LINKED ISSUE #154: Support for resque 2.0
Is there any intention for this gem to support resque 2.0 or higher? I see that this was brought up over 3 years ago in #116, but as of a month ago [2.0.0 has finally been officially released](https://github.com/resque/resque/releases/tag/v2.0.0).

Thanks for the great library!

### PR DESCRIPTION
**Pull request by peret**
Support Resque 2.0. Closes #154.

### CODE REVIEW

**Comment by jzaleski** [comment]
Thanks for starting this. I won't be able to merge this and cut a new gem until the tests have passed however

**Comment by peret** [comment]
Of course! I was a little confused though, because the tests passed locally for me. It appears that the test suite might be depending on the order of test cases, because I can reproduce different results locally with different seed values. You should be able to verify this by running
`rake test TESTOPTS="--seed=6729"` and then `rake test TESTOPTS="--seed=10451"`. The first run passes, the second one does not.

This *also happens in master*, so I'm not sure how to proceed here. As far as I can see, this is not an issue with the branch. Can we just retry the build?

**Comment by jzaleski** [comment]
Odd. I kicked the build. If we have no-deterministic tests we should probably fix that as well. I don't recall seeing this before.

File: test/resque_test.rb [hunk 1]
```diff
@@ -4,8 +4,8 @@
 class ResqueTest < Minitest::Test
   def test_resque_version
     major, minor, patch = Resque::Version.split('.')
-    assert_equal 1, major.to_i, 'major version does not match'
-    assert_operator minor.to_i, :>=, 8, 'minor version is too low'
+    assert_equal 2, major.to_i, 'major version does not match'
```
Comment (jzaleski) [inline test/resque_test.rb]: This should probably be `1` or `2` to ensure cross-compatibility

File: test/resque_test.rb [hunk 2]
```diff
@@ -4,8 +4,8 @@
...
+    assert_operator minor.to_i, :>=, 0, 'minor version is too low'
```
Comment (jzaleski) [inline test/resque_test.rb]: This line should also be re-thought

**Comment by jzaleski** [comment]
It seems like it's going to pass this time. We're just waiting on `jruby`

**Comment by jzaleski** [comment]
Could you also please `rebase` off of the _master_ branch? I just updated the build-matrix for Travis, so that it's on the HEAD version of all of the minor revisions and added `ruby-2.6`

---

I will clean up all the deprecations once this is merged, bump the version and cut a new `gem` -- thanks again!

**Comment by peret** [comment]
Great, thanks! Hope this works now :)

**Comment by jzaleski** [comment]
Looks good. Please update the version in the README.md and I'll get this merged.

Thanks for the contribution

**Comment by jzaleski** [comment]
This is merged.. I just pushed another commit (https://github.com/lantins/resque-retry/commit/4d82c449d3c07da5e40dfa169aa46630a4b0d4bb) to clean up the deprecation warnings on newer `ruby` -- there are still a number of them coming out of `resque` itself and I've definitely seen the issues as described here: https://github.com/lantins/resque-retry/issues/157

I think we're fine to cut a new `gem` now, in the longer term I'd like to fix the issue above. PRs welcome!

**Comment by jzaleski** [comment]
Alrighty! It's out there: https://rubygems.org/gems/resque-retry/versions/1.6.0

**Pull request merged**